pygeoapi/process/manager/base.py

```diff
@@ -0,0 +1,303 @@
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+from datetime import datetime
+import io
+import json
+import logging
+from multiprocessing import dummy
+import os
+
+from pygeoapi.util import DATETIME_FORMAT, JobStatus
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseManager:
+    """generic Manager ABC"""
+
+    def __init__(self, manager_def):
+        """
+        Initialize object
+
+        :param manager_def: manager definition
+
+        :returns: `pygeoapi.process.manager.base.BaseManager`
+        """
+
+        self.name = manager_def['name']
+        self.connection = manager_def['connection']
+        self.output_dir = manager_def.get('output_dir', None)
+
+    def create(self):
+        """
+        Create manager
+
+        :returns: `bool` status of result
+        """
+
+        raise NotImplementedError()
+
+    def connect(self):
+        """
+        Connect manager
+
+        :returns: `bool` status of result
+        """
+
+        raise NotImplementedError()
+
+    def destroy(self):
+        """
+        Destroy manager
+
+        :returns: `bool` status of result
+        """
+
+        raise NotImplementedError()
+
+    def get_jobs(self, process_id=None, status=None):
+        """
+        Get process jobs, optionally filtered by status
+
+        :param process_id: process identifier
+        :param status: job status (accepted, running, successful,
+                       failed, results) (default is all)
+
+        :returns: `list` of jobs (identifier, status, process identifier)
+        """
+
+        raise NotImplementedError()
+
+    def add_job(self, job_metadata):
+        """
+        Add a job
+
+        :param job_metadata: `dict` of job metadata
+
+        :returns: `str` added job identifier
+        """
+
+        raise NotImplementedError()
+
+    def update_job(self, process_id, job_id, update_dict):
+        """
+        Updates a job
+
+        :param process_id: process identifier
+        :param job_id: job identifier
+        :param update_dict: `dict` of property updates
+
+        :returns: `bool` of status result
+        """
+
+        raise NotImplementedError()
+
+    def get_job_status(self, process_id, job_id):
```

Comment:
This is currently unused. In api.py, `get_process_job_status()` actually calls `manager.get_job_result()`, so we should probably unify whether we call this `status` or `result`

Reply:
Fixed.